indigo_app/views/places.py

```diff
@@ -100,8 +100,133 @@ def get_context_data(self, **kwargs):
+        context['subsidiary_works_percentage'] = 100 - context['primary_works_percentage']
+
+        # Completeness
+        metrics = list(DailyWorkMetrics.objects
+            .filter(place_code=self.place.place_code)
+            .filter(date__gte=since)
+            .order_by('date')
+            .all())
+
+        if metrics:
+            context['latest_stat'] = metrics[-1]
+
+        # breadth completeness history
+        context['completeness_history'] = json.dumps([
+            [m.date.isoformat(), m.p_breadth_complete]
+            for m in metrics])
+
+        return context
+
+    def get_recently_updated_works(self):
+        # TODO: this should also factor in documents that have been updated
+        return Work.objects \
+            .filter(country=self.country, locality=self.locality) \
+            .order_by('-updated_at')[:5]
+
+    def get_recently_created_works(self):
+        return Work.objects \
+                   .filter(country=self.country, locality=self.locality) \
+                   .order_by('-created_at')[:5]
+
+    def get_works_by_subtype(self, works):
+        def subtype_name(abbr):
+            if not abbr:
+                return 'Act'
+            st = Subtype.for_abbreviation(abbr)
+            return st.name if st else abbr
+        pairs = list(Counter([subtype_name(w.subtype) for w in works]).items())
+        pairs = [list(p) for p in pairs]
+        pairs.sort(key=lambda p: p[1], reverse=True)
+
+        total = sum([x[1] for x in pairs])
+        for p in pairs:
+            p.append(int((p[1] / (total or 1)) * 100))        
+
+        return pairs
+
+    def calculate_open_tasks(self):
+        tasks = Task.objects.filter(country=self.country, locality=self.locality) \
+            .filter(Q(state=Task.OPEN) | Q(state=Task.PENDING_REVIEW))
+        total_open_tasks = tasks.count()
+        pending_review_tasks = tasks.filter(state='pending_review').count()
+        open_tasks = tasks.filter(state='open').exclude(assigned_to__isnull=False).count()
+        assigned_tasks = tasks.filter(state='open').exclude(assigned_to=None).count()
+
+        task_chart = [{
+                'state': 'open',
+                'state_string': 'Unassigned',
+                'count': open_tasks,
+                'percentage': int((open_tasks / (total_open_tasks or 1)) * 100)
+            },
+            {
+                'state': 'assigned',
+                'state_string': 'Assigned',
+                'count': assigned_tasks,
+                'percentage': int((assigned_tasks / (total_open_tasks or 1)) * 100)
+            },
+            {
+                'state': 'pending_review',
+                'state_string': 'Pending Review',
+                'count': pending_review_tasks,
+                'percentage': int((pending_review_tasks / (total_open_tasks or 1)) * 100)
+            }]
+
+        # open tasks by label
+        labels = TaskLabel.objects.all()
```

Comment:
If you change the `related_name` on `Task.labels` to `tasks`, then you can do this:

```
from django.db.models import Count

labels = TaskLabel.objects \
    .filter(tasks__state__in=Task.OPEN_STATES) \
    .filter(country=self.country, locality=self.locality) \
    .annotate(n_tasks=Count('tasks__id'))
```

Then `labels[0].n_tasks` in the number of open tasks for that label.

Reply:
I had to modify it a little since you can't filter the TaskLabels by country or locality:
```
labels = TaskLabel.objects.filter(tasks__in=tasks) \
            .annotate(n_tasks=Count('tasks__id'))
```
But it worked as expected.

Reply:
Cool. You also could do `filter(tasks__country=...)`, but yours works too.